### PR TITLE
Implement exception handler service; fixes #345

### LINKED ISSUE #345: Implement ExceptionHandlerService in CDP4Common and use in CDP4Dal.Session
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am running the latest version of the COMET-SDK
- [x] I have searched [open](https://github.com/STARIONGROUP/COMET-SDK-Community-Edition/issues) and [closed](https://github.com/STARIONGROUP/COMET-SDK-Community-Edition/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description
Implement ExceptionHandlerService in CDP4Common and use in CDP4Dal.Session

### PR DESCRIPTION
**Pull request by lxatstariongroup**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/STARIONGROUP/COMET-SDK-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-SDK [code style guidelines](https://raw.githubusercontent.com/STARIONGROUP/COMET-SDK-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### CODE REVIEW

**Comment by lxatstariongroup** [comment]
Small comment about implementation:
The system will execute all connected handlers every time, regardless of the result (ishandled) of an individual handler in case multiple handlers are executed. So it does not stop the process when an individual handler returns IsHandled = true.
The reason is that, due to DI registration/resolving of IExceptionHandlers, we don't have control over the order in which the IExceptionHandlers are executed.
Do you agree on that?

File: CDP4Common.Tests/CDP4Common - Backup.Tests.csproj
Comment (antoineatstariongroup) [inline CDP4Common.Tests/CDP4Common - Backup.Tests.csproj:1]: Is this file necessary?

**Review by antoineatstariongroup — APPROVED** [review]
Check file comment

File: CDP4Common.Tests/CDP4Common - Backup.Tests.csproj
Comment (lxatstariongroup) [reply CDP4Common.Tests/CDP4Common - Backup.Tests.csproj:1]: good catch!
